agent/engine/task_manager.go

```diff
@@ -148,6 +149,18 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO: Add new task resources provisioned state ?
+			err := mtask.SetupPlatformResources()
+			if err != nil {
```

Comment:
I'm sure that we need to test this path but we only have stubs with this PR and we don't have a way to obtain a `non-nil` condition here at this moment. 

Changes to api.Task will result in changes to task.CgroupEnabled() and all the stubs will be replaced with actual implementation. I'd like to focus this PR on `resources/cgroup`. 

This PR would be followed up with another with changes to managedTask and api.Task.